idaes/surrogate/keras_surrogate.py

```diff
@@ -0,0 +1,201 @@
+        offset_inputs = np.zeros(self.n_inputs())
+        factor_inputs = np.ones(self.n_inputs())
+        offset_outputs = np.zeros(self.n_outputs())
+        factor_outputs = np.ones(self.n_outputs())
+        if self._input_scaler:
+            offset_inputs = self._input_scaler.offset_series()[self.input_labels()].to_numpy()
+            factor_inputs = self._input_scaler.factor_series()[self.input_labels()].to_numpy()
+        if self._output_scaler:
+            offset_outputs = self._output_scaler.offset_series()[self.output_labels()].to_numpy()
+            factor_outputs = self._output_scaler.factor_series()[self.output_labels()].to_numpy()
+
+        omlt_scaling = OffsetScaling(offset_inputs=scaling_x_offset,
+                                      factor_inputs=scaling_x_factor,
+                                      offset_outputs=scaling_y_offset,
+                                      factor_outputs=scaling_y_factor)
+        net = load_keras_sequential(self._keras_model, omlt_scaling, self.input_bounds())
+
+        if formulation == 'full-space':
+            formulation_object = FullSpaceContinuousFormulation(net)
+        elif formulation == 'reduced-space':
+            formulation_object = ReducedSpaceContinuousFormulation(net)
+        elif formulation == 'relu-bigm':
+            formulation_object = ReLUBigMFormulation(net)
+        elif formulation == 'relu-complementarity':
+            formulation_object = ReLUComplementarityFormulation(net)
+
+        block.neural_network.build_formulation(formulation_object,
+                                               input_vars=block._input_vars_as_list(),
+                                               output_vars=block._output_vars_as_list())
+
+
+
+    def evaluate_surrogate(self, inputs):
+        """
+        Method to evaluate Keras model at a set of input values.
+
+        Args:
+            inputs: numpy array of input values. First dimension of array
+                must match the number of input variables.
+
+        Returns:
+            outputs: numpy array of values for all outputs evaluated at input
+                points.
+        """
+        x = inputs
+        if self._input_scaler is not None:
+            x = self._input_scaler.scale(x)
+            
+        y = self._keras_model.predict(x.to_numpy())
+
+        # y is a numpy array, make it a dataframe
+        y = pd.Dataframe(data=y, columns=self.output_labels(), index=inputs.index)
+
+        if self._output_scaler is not None:
+            y = self._output_scaler.unscale(y)
+
+        return y
+
+    @classmethod
+    def save_to_file(cls, keras_folder_name):
+        """
+        Save the surrogate object to disk by providing the name of the
+        folder to contain the keras model and additional IDAES metadata
+
+        Args:
+           folder_name: str
+              The name of the folder to contain the Keras model and additional
+              IDAES metadata
+        """

```

Comment:
This doesn't need to change for this PR, but I think that long term this method should:
- create a temporary directory, 
- pass that dir to `_keras_model.save()`, 
- add the `idaes_info.json` to the dir,
- zip up the dir to the file name passed to `save_to_file`
- delete the temporary directory

Reply:
I agree. I also found a new way to store and read keras models (used for testing now). We might be able to build off of that.